### ISSUE TITLE
Menu position fixed when scroll

### ISSUE THREAD

**Issue by StevenHeven** [Vuepic/vue-datepicker#253]
**Describe the bug**
Don't understand why, and after multiple searches, when I'm scrolling, the calendar overlay (`.dp__menu`) stay at fixed position and no attached at the field (GIF illustration below). I check if we have overwrite style can do this, but nothing found.

**To Reproduce**
My DatePicker component in TS : 
```
<script setup lang="ts">
import { InputField, Icon } from '../@atoms';
import Datepicker from '@vuepic/vue-datepicker';
import '@vuepic/vue-datepicker/dist/main.css';
import { InputSlotsProps } from '../@atoms/aInputField.vue';
import { endOfMonth, endOfYear, startOfMonth, startOfYear, subMonths, startOfISOWeek, endOfISOWeek, subWeeks, startOfDay } from 'date-fns';

export interface DatePickerProps {
  label: string;
  onlyTimePicker?: boolean;
  needTimePicker?: boolean;
  range?: boolean;
  locale?: string;
  required?: boolean;
  valueDate?: Date | number;
  minDate?: Date | string;
  maxDate?: Date | string;
  minRange?: number;
  maxRange?: number;
}

const props = withDefaults(defineProps<DatePickerProps>(), {
  needTimePicker: true,
  locale: 'fr-FR'
});

const emit = defineEmits<{
  (e: 'update:modelValue', value: Date | number | undefined): Date;
}>();

const date = ref<Date | number>();
const state = ref<boolean>();
const now = new Date();
const presetRanges = ref([
  { label: 'This week', range: [startOfISOWeek(now), endOfISOWeek(now)] },
  { label: 'Last week', range: [startOfISOWeek(subWeeks(now, 1)), endOfISOWeek(subWeeks(now, 1))] },
  { label: 'This month', range: [startOfMonth(now), endOfMonth(now)] },
  {
    label: 'Last month',
    range: [startOfMonth(subMonths(now, 1)), endOfMonth(subMonths(now, 1))]
  },
  {
    label: 'Last sliding month',
    range: [startOfDay(subMonths(now, 1)), now]
  },
  { label: 'This year', range: [startOfYear(now), endOfYear(now)] }
]);

onMounted(() => {
  date.value = props.valueDate;
});

function checkValidation(inputProps: InputSlotsProps) {
  if (props.required) {
    state.value = date.value !== undefined && date.value !== null;
    inputProps.isValid(state.value);
  }
  emit('update:modelValue', date.value);
}

function determineFormat(): string {
  let formatDate = '';
  let formatHour = '';
  let formatGlobal = '';
  if (props.locale === 'fr-FR') {
    formatDate = 'dd/MM/yyyy';
  } else {
    formatDate = 'MM-dd-yyyy';
  }

  formatHour = 'HH:mm';
  if (props.needTimePicker && !props.onlyTimePicker) {
    formatGlobal = formatDate + ' ' + formatHour;
  }
  if (props.onlyTimePicker) {
    formatGlobal = formatHour;
  }

  return formatGlobal;
}
</script>

<template>
  <div class="mol-date-picker">
    <InputField :label="props.label" :required="props.required">
      <template #input="inputProps">
        <Datepicker
          class="date-picker"
          v-model="date"
          textInput
          :class="props.range && 'range'"
          :enable-time-picker="props.needTimePicker"
          :time-picker="props.onlyTimePicker"
          :flow="['calendar', 'time']"
          :format="determineFormat()"
          :locale="props.locale"
          :min-date="props.minDate"
          :max-date="props.maxDate"
          :min-range="props.minRange"
          :max-range="props.maxRange"
          :partial-range="props.range && false"
          :range="props.range"
          :preset-ranges="props.range ? presetRanges : undefined"
          :state="state"
          :auto-position="false"
          @cleared="checkValidation(inputProps)"
          @update:modelValue="checkValidation(inputProps)"
          @blur="checkValidation(inputProps)"
          @open="inputProps.toggleFocus"
          @closed="inputProps.toggleFocus"
        >
          <template #input-icon>
            <Icon
              class="date-time-icon"
              :use="props.onlyTimePicker ? 'iterop-clock-regular' : props.range ? 'iterop-date-range-regular' : 'iterop-calendar-regular'"
            />
          </template>

          <template #yearly="{ label, range, presetDateRange }">
            <span @click="presetDateRange(range)">{{ label }}</span>
          </template>
        </Datepicker>
      </template>
    </InputField>
  </div>
</template>

<style scoped lang="scss">
.mol-date-picker {
  position: relative;

  .date-picker {
    :deep(.dp__input_wrap input) {
      border: none;
    }

    :deep(.dp__input_valid),
    :deep(.dp__input_invalid) {
      box-shadow: none;
    }

    :deep(.dp__input_icon) {
      top: 75%;
      transition: all 0.4s;
    }

    &.range {
      min-width: rem(325);
    }
  }

  .focus .date-time-icon {
    fill: $sea-blue;
  }

  .valid .date-time-icon {
    fill: $grass-green;
  }

  .invalid .date-time-icon {
    fill: $blaze-red;
  }
}
</style>
```

**Expected behavior**
Like your demo, menu stay attached to field when scrolling.

**Screenshots**
![Date Picker positioning gif](https://www.zupimages.net/up/23/04/6yfj.gif)

**Desktop & mobile (please complete the following information):**
 - On all Browser

**Comment by Jasenkoo** [comment]
Very strange, maybe `InputField` slot has something to do with it, can you check if it is still stuck if you place it outside of slot? Also, can you try setting `:alt-position="true"`

**Comment by Jasenkoo** [comment]
I tried with coping all props you used and can't replicate the issue. Closing this as this is caused by something project related. I would suggest you check if there are any errors in the console.

Also, make sure that the scroll listener is not blocked by something.

If not there is an exposed method `onScroll` (just realized this is not documented) you can add a global window scroll listener and call this method.

**Comment by tumur1** [comment]
Same issue here. When scrollable parent is not 'body', menu position is fixed. And "close-on-scroll" is not working.
I forked the playground.
![ss1](https://user-images.githubusercontent.com/123346464/218362001-e6f4df2f-ab70-4c5c-b2f2-b2b8ed3ee722.png)
![ss2](https://user-images.githubusercontent.com/123346464/218362006-63911846-cf04-4cf2-a1b4-fd7aef1f35b6.png)

playground URL:
https://stackblitz.com/edit/vuepic-vue-datepicker-tfy4dy?file=src/components/Playground.vue

**Comment by StevenHeven** [comment]
I'm feeling less alone ^^ and I found no solution yet

**Comment by Jasenkoo** [comment]
@tumur1 Thank you for the demo, will check it out.

**Comment by StevenHeven** [comment]
With your PR, I've no overlay anymore ... instead with clean module, or deleted `node_modules` folder .... 
![bug-date-picker-4.0](https://www.zupimages.net/up/23/07/nf36.gif)

In HTML , we can see the overlay not appear. 
With 3.6.8 version : 
![image](https://user-images.githubusercontent.com/27998167/219608140-6fafee5b-e04c-4d54-81f4-5c9635cf7dcb.png)
With 4.0.0 version : 
![image](https://user-images.githubusercontent.com/27998167/219608219-57e53f74-d6ed-476d-8dbc-b850ba5e4869.png)

Do you any suggestions ?

**Comment by Jasenkoo** [comment]
@StevenHeven Can you inspect and see where it is placed, it should be under the input, just to confirm if it is the wrong placement or z-index property.

**Comment by StevenHeven** [comment]
Or i didn't see but the menu appear at the top of my page xD 
![bug-date-picker](https://zupimages.net/up/23/07/yki1.gif)

I use `:altPosition` to set position .... only with undefined value it's works to get the position attached with overwrite SCSS : 
```
const customPosition = () => ({ top: datePicker.value.top, left: datePicker.value.left });
</script>

<template>
  <div class="mol-date-picker">
    <InputField :label="props.label" :required="props.required" :value="props.valueDate?.toString()">
      <template #input="inputProps">
        <Datepicker
          class="date-picker"
          v-model="date"
          textInput
          ref="datePicker"   <---- datePicker reference
          auto-apply
          keepActionRow
          :altPosition="customPosition"   <---- here
          :dark="themeDatePicker()"
          :class="props.range && 'range'"
          :enable-time-picker="props.needTimePicker"
          :time-picker="props.onlyTimePicker"
          :flow="['calendar', 'time']"
          :format="determineFormat()"
          :locale="props.locale"
          :minDate="props.minDate"
          :maxDate="props.maxDate"
          :minRange="props.minRange"
          :maxRange="props.maxRange"
          :partialRange="props.range && false"
          :range="props.range"
          :preset-ranges="props.range ? presetRanges : undefined"
          :state="state"
          @cleared="checkValidation(inputProps)"
          @update:modelValue="checkValidation(inputProps)"
          @blur="checkValidation(inputProps)"
          @open="inputProps.toggleFocus"
          @closed="inputProps.toggleFocus"
          @keydown.esc="cancelSelection"
        >
          <template #input-icon>
            <Icon class="date-time-icon" :icon="props.onlyTimePicker ? 'clock' : 'calendar'" />
          </template>

          <template #action-row="{ internalModelValue, selectDate }">
            <div class="action-row" style="text-align: center">
              <p class="current-selection">{{ formatPreviewDate(internalModelValue) }}</p>
              <Button :aspect="ButtonAspect.FILLED" @button-click="selectDate" label="Select" />
            </div>
          </template>

          <template #yearly="{ label, range, presetDateRange }">
            <span @click="presetDateRange(range)">{{ label }}</span>
          </template>
        </Datepicker>
      </template>
    </InputField>
  </div>
</template>

<style scoped lang="scss">
   [...]

  .date-picker {
    [...]

    :deep(.dp__menu) {
      margin-top: rem(12);
      margin-left: rem(-40);
    }

 ```
`datePicker.value.top` not exist. 
If I used HTMLElement passed in `altPosition`, `el.offsetTop` or `el.offsetLeft` is always 0 .... 

And I have this error on `altPosition` : 
``` 
Type '() => { top: any; left: any; }' is not assignable to type 'boolean | ((el: HTMLElement | undefined) => { top: string; left: string; transform: string; }) | undefined'.
  Type '() => { top: any; left: any; }' is not assignable to type '(el: HTMLElement | undefined) => { top: string; left: string; transform: string; }'.ts(2322)
```

It works, menu stay at the position, but with strange logic ...

**Comment by Jasenkoo** [comment]
@StevenHeven Tnx, will check what is going on.

**Comment by StevenHeven** [comment]
I tried this : 
`const customPosition = (el: HTMLElement) => ({ top: el.getBoundingClientRect().top, left: el.getBoundingClientRect().left });`

And I've same position ... Maybe the element pass get wrong coordinates position...

**Comment by Jasenkoo** [comment]
> I tried this when I found this method [element.getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) : `const customPosition = (el: HTMLElement) => ({ top: el.getBoundingClientRect().top, left: el.getBoundingClientRect().left });`
> 

When you pass this function it doesn't work properly?

**Comment by StevenHeven** [comment]
> > I tried this when I found this method [element.getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) : `const customPosition = (el: HTMLElement) => ({ top: el.getBoundingClientRect().top, left: el.getBoundingClientRect().left });`
> 
> When you pass this function it doesn't work properly?

Nope, I've the same position like as my GIF demonstrate upper. The value was, by memory, 425 for top and 215 for left, for all DatePiker. It's good left value, not for top :/

**Comment by Jasenkoo** [comment]
Ok, I'll change the positioning, looks like `position: fixed` is having some issues.

**Comment by StevenHeven** [comment]
IT WORKS !! Thx a lot ^^ With no `alt-position` props ^^

**Comment by StevenHeven** [comment]
BUT ... Bug appears for range mode with presets : 
![image](https://user-images.githubusercontent.com/27998167/219712138-e57174c6-ab52-41bc-92b5-f791990cd10b.png)

**Comment by Jasenkoo** [comment]
> IT WORKS !! Thx a lot ^^ With no `alt-position` props ^^

Great

For the layout, please open a new issue.

**Comment by axelkennedal** [comment]
I encountered this same issue, upgrading to the latest version of vue-datepicker fixed it :)
